loop/views.py

```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
+        total_share += entry['amount']
+    return total_share
+
+
+def calculate_gaddidar_share(start_date, end_date, mandi_list, aggregator_list):
+
+    parameters_dictionary = {'mandi__in': mandi_list}
+    parameters_dictionary_for_outliers = {
+        'mandi__in': mandi_list, 'aggregator__user__in': aggregator_list}
+    parameters_dictionary_for_ct = {'date__gte': start_date, 'date__lte': end_date,
+                                    'mandi__in': mandi_list, 'user_created__id__in': aggregator_list}
+
+    arguments_for_ct = {}
+    arguments_for_gaddidar_commision = {}
+    arguments_for_gaddidar_outliers = {}
+
+    for k, v in parameters_dictionary.items():
+        if v:
+            arguments_for_gaddidar_commision[k] = v
+
+    for k, v in parameters_dictionary_for_ct.items():
+        if v:
+            arguments_for_ct[k] = v
+
+    for k, v in parameters_dictionary_for_outliers.items():
+        if v:
+            arguments_for_gaddidar_outliers[k] = v
+
+    gc_queryset = GaddidarCommission.objects.filter(
+        **arguments_for_gaddidar_commision)
+    gso_queryset = GaddidarShareOutliers.objects.filter(
+        **arguments_for_gaddidar_outliers)
+    combined_ct_queryset = CombinedTransaction.objects.filter(**arguments_for_ct).values(
+        'date', 'user_created_id', 'gaddidar', 'mandi', 'gaddidar__discount_criteria').order_by('-date').annotate(Sum('quantity'), Sum('amount'))
+    result = []
+    for CT in combined_ct_queryset:
+        sum = 0
+        user = LoopUser.objects.get(user_id=CT['user_created_id'])
+        if CT['date'] not in [x.date for x in gso_queryset]:
+            try:
+                gc_list_set = gc_queryset.filter(start_date__lte=CT['date'], gaddidar=CT[
+                                                 'gaddidar']).order_by('-start_date')
+                if CT['gaddidar__discount_criteria'] == 0 and len(gc_list_set) > 0:
+                    sum += CT['quantity__sum'] * \
+                        gc_list_set[0].discount_percent
+                elif len(gc_list_set) > 0:
+                    sum += CT['amount__sum'] * gc_list_set[0].discount_percent
+            except GaddidarCommission.DoesNotExist:
+                pass
+        else:
+            try:
+                gso_gaddidar_date_aggregator = gso_queryset.filter(
+                    date=CT['date'], aggregator=user.id, gaddidar=CT['gaddidar']).values_list('amount', flat=True)
+                if len(gso_gaddidar_date_aggregator):
+                    sum += gso_gaddidar_date_aggregator[0]
+            except GaddidarShareOutliers.DoesNotExist:
+                pass
+        result.append({'date': CT['date'], 'user_created__id': CT['user_created_id'], 'gaddidar__id': CT[
+                      'gaddidar'], 'mandi__id': CT['mandi'], 'amount': sum, 'quantity__sum': CT['quantity__sum']})
+    return result
+
+
+def crop_language_data(request):
+    crops = Croplanguage.objects.filter(language=request.GET.get('language'))
```

Comment:
Change the Croplanguage to CropLanguage after making required changes in models.py